STRUCTURE.md

```diff
@@ -0,0 +1,123 @@
+  correct and secure way;
+* Wrappers around raw `ForeignPtr`s for data that is meant to be passed around,
+  or generated in a very specific way.
+
+Specifically, the following principles generally apply:
+
+* These libraries cannot depend on non-boot packages;
+* Abstraction should be kept minimal. `newtype`-wrapping a `ForeignPtr` is fine,
+  type class hierarchies without laws are not.
+* Whenever possible, if there are multiple secure ways to do something, without
+  a definite 'this one is always better' option, users of these libraries should
+  be given _all_ options, as well as a detailed explanation of why they should
+  prefer any given one.
+* Partiality must not exist unless specifically marked as unsafe, through a
+  dedicated `Unsafe` module or prefixing the name with `unsafe`. This extends to
+  instances of known partial type classes (such as `Read`).
+
+These principles apply to 'algorithmic' category packages specifically:
+
+* The packages should document as much as possible that is known about the
+  algorithm, providing relevant links for detail where it would be too hard to
+  include.
+* The typical _secure_ uses should be enumerated.
+* Any caveats to performance should be documented clearly.
+* 'The algorithm is the focus': abstractions should only be done to hide the use
+  of C FFI, or where the abstraction is both obvious and unarguable. Use cases
+  should not be pre-supposed.
+
+These principles apply to 'functional' category packages specifically:
+
+* 'The task is the focus': specifics regarding algorithms or implementations
+  should only factor into the abstractions if there really is no other way.
+* There should be one, and only one, correct way to perform the task the package
+  is designed for. If there are legitimately multiple right ways to do
+  something, their differences must be _clearly_ enumerated and marked, with a
+  focus on non-security-familiar users.
+* Doing the right thing should be easy; doing the wrong thing should be
+  impossible.
+
+## Organization
+
+### What gets bindings and wrappers?
+
+Given infinite time, there are quite a few possible bindings and wrappers we
+_could_ define: however, given finite time and labour, some standards for
+inclusion and effort must be put down. In our case, given any possible thing we
+could work on, it must fit the following criteria to be considered, in the
+following order:
+
+1. Is this a good practice in the current year regarding security?
+2. Is there genuine need from some person or organization in the ecosystem and
+   community for this?
+
+We prioritize security over demand, as one of the core
+foci of this organization and project is the improvement of security-related
+practices in the Haskell ecosystem.
+By implementing functionality that is known
+to be insecure or a bad practice, we are implicitly stating that this is OK,
+which works directly counter to a stated goal.
```

Comment:
:+1: on this, maybe eventually a `legacy` project for outdated stuff would be a fun weekend project but definitely not anything exposed in the main packages.

Reply:
Hmm, yeah, a `legacy` namespace as well would be interesting.